lib/adrian/directory_queue.rb

```diff
@@ -69,9 +68,12 @@ def reserve(item)
     end
 
     def items
-      items = files.map { |file| wrap_item(file) }
-      items.reject! { |item| !item.exist? || filter?(item) }
-      items.sort_by(&:updated_at)
+      return @items if @items && @items.length > 0
+      @items = begin
+        items = files.map { |file| wrap_item(file) }
```

Comment:
can all be done on `@items` without the begin ... maybe map! on files and sort_by!

```Ruby
@items = files.map! { |file| wrap_item(file) }
@items.reject! { |item| filter?(item) } # !move exist? into filter maybe ?
@items.sort_by!(&:updated_at)
```